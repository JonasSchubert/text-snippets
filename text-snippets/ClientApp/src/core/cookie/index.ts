export const appName = "dotnetcore-vue-template";

export const daysTilExpired: { [key: string]: number } = {};

export const keys: { [key: string]: string } = {};
