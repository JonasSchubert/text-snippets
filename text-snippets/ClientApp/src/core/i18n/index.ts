import Vue from 'vue';
import VueI18n from 'vue-i18n';

import enGb from './en-GB';

Vue.use(VueI18n);

export const defaultLocale = 'en-GB';
export const defaultMessages = { [defaultLocale]: enGb };

const i18n = new VueI18n({
  fallbackLocale: defaultLocale,
  locale: defaultLocale,
  messages: defaultMessages,
  silentTranslationWarn: process.env.NODE_ENV === 'production'
});

export default i18n;
