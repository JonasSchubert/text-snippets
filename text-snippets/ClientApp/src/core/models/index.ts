export * from './root-state.d';
