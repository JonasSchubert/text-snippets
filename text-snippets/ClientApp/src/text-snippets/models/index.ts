export * from './text-snippet.d';
export * from './text-snippets-state.d';
