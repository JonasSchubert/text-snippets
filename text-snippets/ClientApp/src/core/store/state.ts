import { RootState } from '@/core/models';

export const createState = (): RootState => ({
  requests: 0
});
