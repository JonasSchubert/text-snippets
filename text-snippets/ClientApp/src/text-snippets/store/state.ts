import { TextSnippetsState } from '@/text-snippets/models';

export const createState = (): TextSnippetsState => ({
  error: undefined,
  isLoading: false,
  list: []
});
