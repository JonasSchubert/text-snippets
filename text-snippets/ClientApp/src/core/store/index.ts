import Vue from 'vue';
import Vuex, { Store, StoreOptions } from 'vuex';
import { createI18nModule } from '@/core/i18n/module-helper';
import { RootState } from '@/core/models';
import { ModuleType as LocalStorageModule } from "@/libs/local-storage/store/types";
import { createLocalStorageModule } from "@/libs/local-storage/store";
import { createModule as createTextSnippetsModule } from '@/text-snippets/store';
import { ModuleType as TextSnippetsModule } from '@/text-snippets/store/types';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

Vue.use(Vuex);

export const createStore = (vuetify: any): Store<RootState> => {
  const store: StoreOptions<RootState> = {
    getters: createGetters(),
    modules: {
      i18n: createI18nModule(),
      [LocalStorageModule]: createLocalStorageModule<RootState>('text-snippets', vuetify),
      [TextSnippetsModule]: createTextSnippetsModule()
    },
    mutations: createMutations(),
    state: createState(),
    strict: process.env.NODE_ENV !== 'production'
  };

  return new Store<RootState>(store);
};
