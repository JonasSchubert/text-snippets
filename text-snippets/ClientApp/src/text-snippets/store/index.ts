import { Module } from 'vuex';
import { RootState } from '@/core/models';
import { TextSnippetsState } from '@/text-snippets/models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

export const createModule = (): Module<TextSnippetsState, RootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(),
  mutations: createMutations(),
  state: createState()
});
