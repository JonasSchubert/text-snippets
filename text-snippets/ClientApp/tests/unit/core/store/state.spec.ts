import { RootState } from '@/core/models';
import { createState } from '@/core/store/state';

describe('Rootstate', () => {
  test('should not change unexpected', () => {
    // Arrange && Act
    const rootState: RootState = createState();

    // Assert
    expect(rootState).toMatchSnapshot();
  });
});
