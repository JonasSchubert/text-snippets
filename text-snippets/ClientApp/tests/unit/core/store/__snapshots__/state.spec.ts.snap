// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rootstate should not change unexpected 1`] = `
Object {
  "requests": 0,
}
`;
